lib/src/idg/seq.rs

```diff
@@ -0,0 +1,180 @@
+use crate::err::Error;
+use crate::kvs::{Key, Val};
+use roaring::RoaringTreemap;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+
+// Id is a unique monotonic id generated by the sequence generator.
+pub(crate) type Id = u64;
+
+// Seq is a sequence generator that can be used to generate unique monotonic ids.
+// It can reuse freed ids by keeping track of them in a roaring bitmap.
+// It is used to generate ids for any SurrealDB objects that need aliases (e.g. namespaces, databases, tables, indexes, etc.)
+#[derive(Clone)]
+pub struct Seq {
+	state_key: Key,
+	available_ids: Option<RoaringTreemap>,
```

Comment:
Thanks for the code comments! 

I do understand the use case of freed ids, but just wanted to highlight that reusing IDs might lead to non-monotonic behavior. That is, if an entity with a higher ID is deleted and its ID is reused, newer objects might end up having IDs lower than previously created ones. This can be quite confusing because the definition says unique monotonic ids, but if we are returning from the freed ids, it isn't really monotonic right?

Or we'd rather say just unique ids?

Reply:
Ahh good point! It turns out I was quite confused. So to summarize, I think we just want an unique ID generator with stable performance, not monotonic one. I'd update the code comment to reflect that...
Thanks again for your insight @arriqaaq!

Reply:
Awesome!

Reply:
@arriqaaq I'd appreciate it if you could take another look! 69d8aa7e1c75f66e342aab8180348e0bc527c842

Reply:
Just did! Thanks for the fix